### PR TITLE
feat: Add react native example (T-2702)

### PR DESCRIPTION
**Pull request by desfero**
A basic `react-native` example with authentication support
image

- A react-native example app with authentication support
- Splitting `IStorageProvider` by `IStorageProvider` and `IObservableStorageProvider` to make sure that someone knows what he's doing when implementing the `subscribe` function 
- Exported some missing parts of the storage from `@lens-protocol/react`

### CODE REVIEW

File: packages/domain/src/use-cases/wallets/ActiveWallet.ts [hunk 1]
```diff
@@ -33,7 +33,11 @@ export class ActiveWallet {
   async requireActiveWallet(): Promise<Wallet> {
     const wallet = await this.getActiveWallet();
 
-    invariant(wallet, 'Active wallet should be defined');
+    invariant(
+      wallet,
+      `Active wallet is not defined.
+       Make sure that the storage configuration (IStorageProvider) properly persists saves`,
```
Comment (desfero) [inline packages/domain/src/use-cases/wallets/ActiveWallet.ts:39]: Have seen on discord that someone was playing with the sdk and got cryptic errors when storage was not properly configured

File: pnpm-workspace.yaml [hunk 2]
```diff
@@ -1,3 +1,5 @@
 packages:
   - "packages/*"
-  - "examples/*"
+  - "examples/web-wagmi"
+  - "examples/shared"
+  - "examples/nextjs"
```
Comment (desfero) [inline pnpm-workspace.yaml:5]: at some point, it's also worth removing `next.js` from the root monorepo

File: examples/react-native/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+It is not published or released anywhere directly.
+
+## Getting Started
+
+Install example dependencies via (should be run from the `examples/react-native` folder:
+```bash
+pnpm install
+```
+
+Make sure your development environment is ready to start locally `react-native` app by following the [official docs](https://reactnative.dev/docs/environment-setup).
```
Comment (reecejohnson) [inline examples/react-native/README.md:16]: May be worthwhile to include some detail around the basic RN setup here. 

I had some issues relating to the fact I didn't have pods installed

**Comment by reecejohnson** [comment]
working locally for me 🫡

something to note: i was using dark mode, by default, and nothing was initially visible: 

image

**Review by krzysu — APPROVED** [review]
I was able to run it locally, but it's a lot of steps to install all dependencies.
I'm a react native novice so I don't know much about the setup here, but it works, so let's just go with this one. 😃 🚀

File: examples/react-native/README.md (hunk 3 above)
Comment (desfero) [reply examples/react-native/README.md:80]: fixed

Comment (krzysu) [reply examples/react-native/README.md:80]: The readme is good, let's release it and wait for feedback, then we will see what are the most common issues and we will improve the readme based on that

**Pull request merged**